modules/history/src/history-helper.js

```diff
@@ -0,0 +1,186 @@
+const SerializableClass = require("abitbol-serializable");
+const self = require("../index.js");
+
+const historyHelper = {
+
+    /**
+     * Check the differences between two state and apply the changes of snapshotState
+     * to currentState
+     * @param  {Object} currentState - The current state of the history
+     * @param  {Object} snapshotState - The new state of the history
+     */
+    applySnapshotDifference: (currentState, snapshotState) => { // eslint-disable-line
+        const {
+            dataStore,
+        } = self.app.modules;
+        // We make a flat object list for each object
+        const currentEntities = historyHelper.flattenSnapshot(currentState);
+        const snapshotEntities = historyHelper.flattenSnapshot(snapshotState);
+
+        Object.keys(currentEntities).forEach(
+            (currentEntityKey) => {
+                let entityFound = false;
+                Object.keys(snapshotEntities).forEach(
```

Comment:
What for ? To avoid line 26/27 ?

```
const currentEntity = currentEntities[currentEntityKey];
const snapshotEntity = snapshotEntities[snapshotEntityKey];
```

Reply:
only for this reason, not a big deal

Reply:
Won't it just change those two lines to :
```js
const currentEntity = currentEntities[1];
const snapshotEntity = snapshotEntities[1];
``` 
?